contracts/src/NativeTokenDestination.sol

```diff
@@ -0,0 +1,313 @@
+    "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/INativeMinter.sol";
+import {INativeTokenDestination} from "./interfaces/INativeTokenDestination.sol";
+import {TeleporterOwnerUpgradeable} from "@teleporter/upgrades/TeleporterOwnerUpgradeable.sol";
+// We need IAllowList as an indirect dependency in order to compile.
+// solhint-disable-next-line no-unused-import
+import {IAllowList} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IAllowList.sol";
+import {IWrappedNativeToken} from "./interfaces/IWrappedNativeToken.sol";
+import {SendTokensInput} from "./interfaces/ITeleporterTokenBridge.sol";
+import {SafeWrappedNativeTokenDeposit} from "./SafeWrappedNativeTokenDeposit.sol";
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @notice Implementation of the {INativeTokenDestination} interface.
+ *
+ * @dev This contract pairs with exactly one `TokenSource` contract on the source chain and one wrapped native
+ * token on the chain this contract is deployed to.
+ * It mints and burns native tokens on the destination chain corresponding to locks and unlocks on the source chain.
+ */
+contract NativeTokenDestination is
+    TeleporterOwnerUpgradeable,
+    INativeTokenDestination,
+    TeleporterTokenDestination
+{
+    using SafeWrappedNativeTokenDeposit for IWrappedNativeToken;
+
+    /**
+     * @notice The address where the burned transaction fees are credited.
+     *
+     * @dev Defined as BLACKHOLE_ADDRESS at
+     * https://github.com/ava-labs/subnet-evm/blob/v0.6.0/constants/constants.go
+     * C-Chain value found at https://github.com/ava-labs/coreth/blob/v0.13.2/constants/constants.go
+     * It is a system-reserved address by default in subnet-evm and coreth, and transfers cannot be sent here manually.
+     */
+    address public constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
+
+    /**
+     * @notice Address used by this contract to blackhole funds, effectively burning them.
+     *
+     * @dev Native tokens are burned by this contract by sending them to this address when transferring tokens back to
+     * the source chain. Different from BURNED_TX_FEES_ADDRESS so that the total amount burned in transaction fees and
+     * the total amount burned to be sent back to the source chain can be tracked separately.
+     * This address was chosen arbitrarily.
+     */
+    address public constant BURN_FOR_TRANSFER_ADDRESS = 0x0100000000000000000000000000000000010203;
+
+    /**
+     * @notice Address used by the source chain to blackhole funds, effectively burning them.
+     *
+     * @dev When reporting burned transaction fee amounts, this address is used as the recipient
+     * address for the funds to be sent to be burned on the source chain.
+     * This address was chosen arbitrarily.
+     */
+    address public constant SOURCE_CHAIN_BURN_ADDRESS = 0x0100000000000000000000000000000000010203;
```

Comment:
Is there any advantage to having a separate address for BURN_FOR_TRANSFER_ADDRESS and SOURCE_CHAIN_BURN_ADDRESS?

Reply:
It basically just allows us to more easily make them different if we need to. They're technically two different things, and had feedback from Cam and I think Matt that they preferred we treat them as separate.